### DISCUSSION TITLE
How to send a FastAPI request on button click.

### DISCUSSION THREAD

**Discussion by ipa-rar** [zauberzeug/nicegui#1440] [Q&A] [answered]
### Question

I am developing a FastAPI + NiceGUI application and I am in a stage where I create API requests for button clicks. The problem is I am able to test the FastAPI with requests and it works fine but when I send the same request like I did here then nothing really happens.

This is the snippet that I need to run on the click of the button
```py
    def save_parameters(payload):
        response = requests.post(f"{API_URL}/setparams", json=payload)
        print(response.status_code)
        print(response.json())
```
This is my current code of button click.
```py
ui.button('Save parameters', on_click=lambda: save_parameters(payload)).classes('items-center')
```

I want to know If this approach is correct or are there better way to do this

I have the following routes in my application
```
**/** - home page
**/ui** - Nicegui front end app with buttons
**/setparams** - To save the parameters that is saved on button click.
```
As far as I understand I am trying to send a request from /ui to /setparams. Does it work this way?

**Comment by rodja** [comment, marked as answer]
All your UI events are already processed by the backend. There is no need to trigger a request on button click. Just call the method which you have decorated with `@app.post('setparams')` in the `on_click` handler.
